File: src/test/java/de/fraunhofer/aisec/codyze/crymlin/RegressionTests.java

package de.fraunhofer.aisec.codyze.crymlin;

import de.fraunhofer.aisec.codyze.analysis.structures.Finding;
import org.junit.jupiter.api.Test;

import java.util.Set;

class RegressionTests extends AbstractMarkTest {

	/**
	 * Test for arguments of NestedConstructExpressions (Java).
	 *
	 * @throws Exception
	 */
	@Test
	void testNestedConstructExpressionsJava() throws Exception {
		Set<Finding> findings = performTest("unittests/regression/nested_constructors/NestedConstructor.java", "unittests/regression/nested_constructors/");
		expected(findings,
			"line 8: Rule PublicKeyInstanceOfVerifier violated");
	}

	/**
	 * Test for arguments of NestedConstructExpressions (C++).
	 *
	 * @throws Exception
	 */
	@Test
	void testNestedConstructExpressionsCpp() throws Exception {
		Set<Finding> findings = performTest("unittests/regression/nested_constructors/nested_constructor.cpp", "unittests/regression/nested_constructors/");
		expected(findings,
			"line 31: Rule PublicKeyInstanceOfVerifier violated");
	}
}

File: src/codyze_legacy/src/test/java/de/fraunhofer/aisec/codyze/crymlin/RegressionTests.kt
package de.fraunhofer.aisec.codyze.crymlin

import java.lang.Exception
import kotlin.Throws
import org.junit.jupiter.api.Test

internal class RegressionTests : AbstractMarkTest() {
    /**
     * Test for arguments of NestedConstructExpressions (Java).
     *
     * @throws Exception
     */
    @Test
    @Throws(Exception::class)
    fun testNestedConstructExpressionsJava() {
        val findings =
            performTest(
                "legacy/unittests/regression/nested_constructors/NestedConstructor.java",
                "legacy/unittests/regression/nested_constructors/"
            )
        expected(findings, "line 8: Rule PublicKeyInstanceOfVerifier violated")
    }

    /**
     * Test for arguments of NestedConstructExpressions (C++).
     *
     * @throws Exception
     */
    @Test
    @Throws(Exception::class)
    fun testNestedConstructExpressionsCpp() {
        val findings =
            performTest(
                "legacy/unittests/regression/nested_constructors/nested_constructor.cpp",
                "legacy/unittests/regression/nested_constructors/"
            )
        expected(findings, "line 31: Rule PublicKeyInstanceOfVerifier violated")
    }
}
